SLACK-GUIDELINES.md

```diff
@@ -128,6 +128,16 @@ in the traditional sense.
 In the case that certain channels have rules or guidelines, they will be listed
 in the purpose or pinned docs of that channel.
 
+## Threaded Conversations
+We believe Slack threads in public channels are extremely useful for quickly and openly collaborating with your fellow contributors.  
+
+
+However, Slack threads are extremely ephemeral, discoverability and searchability of threads isn't nearly as good as alternatives like GitHub Issues, PRs and Github Discussions. For this reason it's important that Slack discussions make their way to less ephemeral and more comprehensive artifacts for sharing ideas and proposals with the wider community.
+
+These artifacts include, but are not limited to: Github Issues + Pull Requests, Github Discussions, Google Docs, or Mailing List posts.
```

Comment:
It's weird to reiterate the set of "more discoverable" locations between this and the previous block. I took out the specific options in the previous paragraph.

I'm also wondering if this should be a stand-alone paragraph, or part of the previous one.

Reply:
Split it out for readability, have no strong opinions one way or the other

Reply:
Would it make sense to add links to the Google Team Drive for Google Docs and the knative-dev googlegroups to make the pointers more concrete ?